src/context/alert.tsx

```diff
@@ -45,93 +46,134 @@ export enum ERROR_MESSAGE {
   PAST_TRANSACTIONS_ERROR = "We are currently facing server issues. Try again later or contact your in-charge if the problem persists."
 }
 
-const defaultAlertProps: AlertModalProps = {
-  alertType: "ERROR",
-  title: "",
-  buttonTexts: {
-    primaryActionText: "OK"
-  },
-  visible: false,
-  onOk: () => {},
-  onCancel: () => {},
-  onExit: () => {}
+export const getTranslationKeyFromError = (error: Error): string => {
+  switch (error.name) {
+    case "LoginError":
+      return "systemErrorLoginIssue";
+    case "LoginLockedError":
+      return "disabledAccess";
+    case "AuthTakenError":
+      return "alreadyUsedQRCode";
+    case "AuthExpiredError":
+    case "AuthNotFoundError":
+      return "expiredQR";
+    case "AuthInvalidError":
+      return "wrongFormatQRScanAgain";
+    case "OTPWrongErrorLastTry":
+      return "invalidInputOTPOneMoreInvalid";
+    case "OTPWrongError":
+      return "invalidInputOTP";
+    case "OTPExpiredError":
+      return "expiredOTP";
+    case "QuotaError":
+      return "systemErrorConnectivityIssues";
+    default:
+      // TODO: Determine fallback value
+      return error.message;
+  }
 };
 
-export const defaultWarningProps: AlertModalProps = {
-  alertType: "WARN",
-  title: "",
-  buttonTexts: {
-    primaryActionText: "Back",
-    secondaryActionText: "Confirm"
-  },
-  visible: false,
-  onOk: () => {},
-  onCancel: () => {},
-  onExit: () => {}
+export const getTranslationKeyFromMessage = (message: string): string => {
```

Comment:
Convert switch cases to object so query is O(1)

Reply:
How does changing the switch cases from Object to string make it O(1)?

Reply:
No what I meant is

```
switch (xxx):
   case 1:
   case 2:
   case 3:
   ...
   case xxx: // worst case O(N) 
```

vs 

```
const obj = {... got n fields ...}
obj[xxx] // O(1)
```

Reply:
After investigating and discussing, we did not manage to find an elegant solution to this. 

A possible solution would be to create a function that searches for the specific error message in `ERROR_MESSAGE` and map the error message to a specific translation key, but that in of itself would potentially be O(n), defeating the purpose of creating that map.

Another possible solution would be to maintain a separate error message to translation key, but that would introduce more tech debt in the future.

This can be looked into in the future if it becomes an issue.

Reply:
This is done for both mappings from `Error` name to translation keys and `Error` messages to translation keys!